### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by Nelia11**
Add Firebase configuration; 
Add Authentication functionality; 
Add components to Signup, Login and restore Password; 
Add UI for Signup and restore Password pages

### CODE REVIEW

File: frontend/src/pages/dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import useLogout from '../../hooks/useLogout';
+
+const Dashboard: React.FC = () => {
+  const logout = useLogout();
+
+  const handleLogout = async (): Promise<void> => {
+    await logout();
+    window.location.href = '/login';
```
Comment (YurijPaharev) [inline frontend/src/pages/dashboard/Dashboard.tsx]: I think you should use react-router for that

File: package.json [hunk 2]
```diff
@@ -18,5 +18,10 @@
   },
   "devDependencies": {
     "concurrently": "^5.1.0"
+  },
+  "dependencies": {
+    "firebase": "^10.4.0",
+    "i": "^0.3.7",
+    "npm": "^10.1.0"
```
Comment (YurijPaharev) [inline package.json:25]: do you need these two dependencies?

**Review by YurijPaharev — COMMENTED** [review]
Wonderful work, I really like how you worked with Zustand, please review some of my comments

**Comment by Nelia11** [comment]
@YurijPaharev thank you a lot for the reviewing our PR!

**Review by YurijPaharev — APPROVED** [review]

**Pull request merged**